platform/darwin/src/MGLOfflineStorage.mm

```diff
@@ -74,7 +78,8 @@ - (void)unpauseFileSource:(__unused NSNotification *)notification {
 - (void)setDelegate:(id<MGLOfflineStorageDelegate>)newValue {
     _delegate = newValue;
     if ([self.delegate respondsToSelector:@selector(offlineStorage:URLForResourceOfKind:withURL:)]) {
-        _mbglFileSource->setResourceTransform([offlineStorage = self](auto kind_, std::string&& url_) -> std::string {
+        delete _mbglResourceTransform;
+        _mbglResourceTransform = new mbgl::Actor<mbgl::ResourceTransform>(*mbgl::util::RunLoop::Get(), [offlineStorage = self](auto kind_, const std::string&& url_) -> std::string {
```

Comment:
@frederoni I will need your help with this one. For instance, I just followed the "code style" here, but I would totally use `std::unique_ptr` if I could, like it was done for Android.